jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1429,6 +1429,14 @@ public void reset()
             _trailers = null;
             _contentBytesWritten = 0;
             _request.getHttpChannelState().resetResponse();
+
+            HttpConfiguration httpConfiguration = _request.getHttpChannelState().getHttpConfiguration();
+            if (httpConfiguration.getSendServerVersion())
+                getHeaders().put(SERVER_VERSION);
+            if (httpConfiguration.getSendXPoweredBy())
+                getHeaders().put(POWERED_BY);
+            if (httpConfiguration.getSendDateHeader())
+                getHeaders().put(_request.getConnectionMetaData().getConnector().getServer().getDateField());
```

Comment:
This should not be necessary, as these fields are all persistent and thus cannot be removed by the reset.